packages/vite/src/node/server/pluginContainer.ts

```diff
@@ -512,13 +514,15 @@ class EnvironmentPluginContainer {
     if (this._closed) return
     this._closed = true
     await Promise.allSettled(Array.from(this._processesing))
+    const config = this.environment.getTopLevelConfig()
     await this.hookParallel(
       'buildEnd',
       (plugin) => this._getPluginContext(plugin),
       () => [],
       (plugin) =>
         this.environment.name === 'client' ||
-        plugin.perEnvironmentStartEndDuringDev !== true,
+        config.server.perEnvironmentStartEndDuringDev ||
+        plugin.perEnvironmentStartEndDuringDev === true,
```

Comment:
I guess this was opposite.

Reply:
boolean conditions are the hardest ones 🤦🏼